### PR TITLE
cmake(bugfix): sync CMake SIM Toolchain file

### PR DESCRIPTION
**Pull request by xuxin930**
## Summary

1. Synchronize CMake's SIM arch Toolchain file and Makefile versions to be consistent

## Impact

Synchronous build system, not a new feature

## Testing

All sim build

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



**No, this PR does not fully meet the NuttX requirements.**

**Here's why and how to improve it:**

* **Summary:**
    *  **Missing "Why"**:  The summary states WHAT is being changed but not WHY.  Is there a specific problem caused by the versions being out of sync? Does this improve maintainability?  Be explicit about the benefit.
* **Impact:**
    * **Too Brief:** While stating "Synchronous build system, not a new feature" is a start, it lacks detail.  Even if no user-facing changes occur, be thorough:
        * **Build Impact:** Does this change any build flags, dependencies, or the build process itself?
        * **Hardware:** Explicitly confirm "NO" impact on any hardware architectures or boards.
        * **Documentation:**  Does this change necessitate any updates to build instructions or documentation?  Even if "NO", state it.
        * **Compatibility:**  Be explicit about backward and forward compatibility.  Does this change affect how the build system interacts with older NuttX versions or tools?
* **Testing:**
    * **Insufficient Detail:**  "All sim build" is not descriptive enough. 
        * **Host Details:** List the specific operating system(s) and compiler(s) used for building. 
        * **Target Details:** Specify the simulator(s) used (e.g., QEMU, Renode) and any relevant configurations.
        * **Logs:** The testing logs sections are empty. Provide actual logs or output demonstrating the issue before the change and the successful outcome after the change.

**Example Improvements**

**Summary:**

> **Synchronize CMake and Makefile Toolchain Versions for SIM Arch**
>
> This change ensures version consistency between the CMake and Makefile-based build systems for the SIM (simulator) architecture, preventing potential build issues and improving maintainability.

**Impact:**

> * **New Feature:** NO
> * **User Impact:** NO
> * **Build Impact:** NO. The build process and required flags remain unchanged.
> * **Hardware Impact:** NO. This change only affects the simulator build environment.
> * **Documentation Impact:** NO. No documentation updates are required.
> * **Security Impact:** NO
> * **Compatibility Impact:** 
>     * **Backward Compatibility:** YES. This change is compatible with previous NuttX versions.
>     * **Forward Compatibility:**  N/A
> * **Other Considerations:** None.

**Testing:**

> **Build Host:**
>     * OS: Ubuntu 20.04
>     * Compiler: GCC 9.4.0
>
> **Target:**
>     * Architecture: SIM
>     * Simulator: QEMU (qemu-system-arm)
>
> **Testing Logs Before Change:**
> ```
>  [Provide actual build logs showing inconsistency or errors]
> ```
>
> **Testing Logs After Change:**
> ```
>  [Provide build logs demonstrating successful, consistent builds]
> ```

File: arch/sim/src/cmake/Toolchain.cmake [hunk 1]
```diff
@@ -18,6 +18,17 @@
...
+add_compile_options(
+  -U_AIX
+  -U_WIN32
+  -U__APPLE__
+  -U__FreeBSD__
+  -U__NetBSD__
+  -U__linux__
+  -U__sun__
+  -U__unix__
+  -U__ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__)
```
Comment (anchao) [inline arch/sim/src/cmake/Toolchain.cmake:39]: unify the changes
https://github.com/apache/nuttx/blob/master/arch/sim/CMakeLists.txt#L68-L93

Comment (xuxin930) [reply arch/sim/src/cmake/Toolchain.cmake:39]: @anchao OK, let's unify the SIM option into Toolchain file and remove setting in SIM board cmakelists.txt

Comment (xuxin930) [reply arch/sim/src/cmake/Toolchain.cmake:50]: Done

File: arch/sim/src/cmake/Toolchain.cmake [hunk 2]
```diff
@@ -70,18 +81,10 @@ if(CONFIG_STACK_USAGE_WARNING)
   add_compile_options(-Wstack-usage=${CONFIG_STACK_USAGE_WARNING})
 endif()
 
-if(CONFIG_ARCH_COVERAGE)
+if(CONFIG_ARCH_COVERAGE_ALL)
```
Comment (xiaoxiang781216) [inline arch/sim/src/cmake/Toolchain.cmake:84]: not used anymore

File: arch/sim/src/cmake/Toolchain.cmake [hunk 3]
```diff
@@ -70,18 +81,10 @@ if(CONFIG_STACK_USAGE_WARNING)
...
   add_compile_options(-fprofile-generate -ftest-coverage)
 endif()
 
-if(CONFIG_SIM_GCOV_ALL)
```
Comment (xiaoxiang781216) [inline arch/sim/src/cmake/Toolchain.cmake]: why remove

File: arch/sim/src/cmake/Toolchain.cmake (hunk 2 above)
Comment (xuxin930) [reply arch/sim/src/cmake/Toolchain.cmake]: Done

File: arch/sim/src/cmake/Toolchain.cmake (hunk 3 above)
Comment (xuxin930) [reply arch/sim/src/cmake/Toolchain.cmake]: Recoverd

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by raiden00pl** [comment]
@xuxin930 hi, could you look at this issue https://github.com/apache/nuttx/issues/14413 ? it looks like this PR is to blame

**Comment by xuxin930** [comment]
> @xuxin930你好，你能看看这个问题#14413吗？看起来这个 PR 是罪魁祸首

@raiden00pl sure I‘ll lcompiled a 14 to test it

**Pull request merged**